client/reader/stream/index.jsx

```diff
@@ -71,6 +77,59 @@ function getDistanceBetweenRecs() {
 	return distance;
 }
 
+/**
+ * Check if two postKeys are for the same siteId or feedId
+ *
+ * @param {Object} postKey1 First post key
+ * @param {Object} postKey2 Second post key
+ * @returns {Boolean} Returns true if two postKeys are for the same siteId or feedId
+ */
+function sameSite( postKey1, postKey2 ) {
+	return postKey1 && postKey2 &&
+		! postKey1.isRecommendationBlock && ! postKey2.isRecommendationBlock && (
+			( postKey1.blogId && postKey1.blogId === postKey2.blogId ) ||
+			( postKey1.feedId && postKey1.feedId === postKey2.feedId )
+		);
+}
+
+/**
+ * Takes two postKeys and combines them into a ReaderCombinedCard postKey.
+ * Note: This only makes sense for postKeys from the same site
+ *
+ * @param {Object} postKey1 must be either a ReaderCombinedCard postKey or a regular postKey
+ * @param {Object} postKey2 can only be a regular postKey. May not be a combinedCard postKey or a recommendations postKey
+ * @returns {Object} A ReaderCombinedCard postKey
+ */
+function combine( postKey1, postKey2 ) {
+	if ( ! postKey1 || ! postKey2 ) {
+		return null;
+	}
+
+	return {
+		isCombination: true,
+		blogId: postKey1.blogId,
+		feedId: postKey1.feedId,
+		index: postKey1.index,
+		postIds: [
+			...( postKey1.postIds || [ postKey1.postId ] ),
+			...( postKey2.postIds || [ postKey2.postId ] ),
+		],
+	};
+}
+
+const combineCards = ( postKeys ) => postKeys.reduce(
+	( accumulator, postKey ) => {
+		const lastPostKey = last( accumulator );
+		if ( sameSite( lastPostKey, postKey ) && ( ! lastPostKey.postIds || lastPostKey.postIds.length < MAX_COMBINED_CARD_POSTS ) ) {
```

Comment:
if we aren't including this on site-stream we can nix the: ` ( ! lastPostKey.postIds || lastPostKey.postIds.length < MAX_COMBINED_CARD_POSTS )`